### PR TITLE
Source selector

### LINKED ISSUE #80: Wire up source selector/data store/diagrams drawing mechanism

### PR DESCRIPTION
**Pull request by hexrcs**
Closes #80

### CODE REVIEW

**Comment by hexrcs** [comment]
@pm8008 As I understand the `GET_COLL_GROUP_DATA` "cmd" grabs and populates the local data store  based on currently selected "base name" right? There doesn't seem to be a way to specify which data collections are requested.

File: frontend/src/libs/wsutils.js [hunk 1]
```diff
@@ -47,37 +45,41 @@ const initHandlers = socket => {
...
-      dataStore.availableCollections = msg.par;
-      break;
+    // // deprecated
+    // case 'LIST_COLL':
+    //   // msg.par will be array
+    //   dataStore.sourceOptions = msg.par;
+    //   break;
     case 'LIST_COLL_GROUPS':
-      // array of array of strings
+      dataStore.sourceOptions = msg.par.length && msg.par.map(x => x[0]);
```
Comment (hexrcs) [inline frontend/src/libs/wsutils.js]: @pm8008 "Source Options" is updated here.

**Comment by pm8008** [comment]
> @pm8008 As I understand the `GET_COLL_GROUP_DATA` "cmd" grabs and populates the local data store based on currently selected "base name" right? There doesn't seem to be a way to specify which data collections are requested.

The corresponding request is getCollectionGroupData(); I've just added the necessary parameter here:
https://github.com/DHSTTOS/implementation/pull/84

**Comment by hexrcs** [comment]
Tested and self merged 🙂

**Pull request merged**